dpe/runtime_dpe_dpe_fuzz_harness.rs

```diff
@@ -0,0 +1,98 @@
+// Licensed under the Apache-2.0 license
+
+use log::{trace, LevelFilter};
+use simplelog::{Config, WriteLogger};
+use std::fs::OpenOptions;
+
+use crypto::OpensslCrypto;
+use dpe::dpe_instance::DpeEnv;
+use dpe::{commands::Command, response::Response, DpeInstance, Support};
+use platform::{DefaultPlatform, AUTO_INIT_LOCALITY};
+
+// TODO: Review impact.
+const SUPPORT: Support = Support {
+    simulation: true,
+    extend_tci: true,
+    auto_init: true,
+    tagging: true,
+    rotate_context: true,
+    x509: true,
+    csr: true,
+    is_ca: true,
+    is_symmetric: true,
+    internal_info: true,
+    internal_dice: true,
+};
+
+struct SimEnv {
+    platform: DefaultPlatform,
+    crypto: OpensslCrypto,
+}
+
+impl DpeEnv for SimEnv {
+    type Crypto = OpensslCrypto;
+    type Platform = DefaultPlatform;
+
+    fn crypto(&mut self) -> &mut OpensslCrypto {
+        &mut self.crypto
+    }
+
+    fn platform(&mut self) -> &mut DefaultPlatform {
+        &mut self.platform
+    }
+}
+
+// One thing to note: AFL is using persistent mode. If objects aren't dropped, this might double-initialise
```

Comment:
Can you expand on what this means?

Reply:
According to cargo-afl - I haven't confirmed - rather than `fork()`-ing, which has a performance penalty, the target function is re-executed instead. Since the `fuzz!` macro expands inside a function, I had been concerned if it was effectively a `while (true)` loop, then the DPE object might still be instantiated, with its old state from previous calls still resident. This would mean that we would also be fuzzing the effect of successive DPE commands, which might be interesting, but unplanned. After thinking about this some more, I don't think that's an issue: since this helper function is used, we are actually returning once, so I think this means that the object will be dropped.

Generally, I've looked further into the internals of the `libfuzzer-sys` crate than `afl`.

Reply:
Ok, sounds good. Thanks for the explanation!